my_game.py

```diff
@@ -875,14 +874,24 @@ def on_update(self, delta_time):
         Movement and game logic
         """
 
+        #Ufo shooting
+        for ufo in self.ufo_list:
+            # Timer for when ufo should shoot
+            ufo.shoot_timer -= delta_time
```

Comment:
The countdown of the timer should be a part of the Ufo's _on_update(delta_time)_. Change to _on_update()_ if we currently use _update()_.

Reply:
Yes the problem was that ufo.update() didn't have delta time, but changing it to on_update fixes that?